src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/EndpointsApiController.java

```diff
@@ -0,0 +1,101 @@
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.openapi;
+
+import lombok.extern.slf4j.Slf4j;
+import org.niis.xroad.restapi.converter.EndpointConverter;
+import org.niis.xroad.restapi.openapi.model.Endpoint;
+import org.niis.xroad.restapi.service.ClientNotFoundException;
+import org.niis.xroad.restapi.service.EndpointService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * Endpoints api
+ */
+@Controller
+@RequestMapping("/api")
+@Slf4j
+@PreAuthorize("denyAll")
+public class EndpointsApiController implements EndpointsApi {
+
+    private final EndpointService endpointService;
+    private final EndpointConverter endpointConverter;
+
+    private static final String NOT_FOUND_ERROR_MSG = "Endpoint not found with id";
+
+    @Autowired
+    public EndpointsApiController(
+            EndpointService endpointService,
+            EndpointConverter endpointConverter) {
+        this.endpointService = endpointService;
+        this.endpointConverter = endpointConverter;
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('VIEW_ENDPOINT')")
+    public ResponseEntity<Endpoint> getEndpoint(String id) {
+        Endpoint endpoint;
+        try {
+            endpoint = endpointConverter.convert(endpointService.getEndpoint(id));
+        } catch (EndpointService.EndpointNotFoundException e) {
+            throw new ResourceNotFoundException(NOT_FOUND_ERROR_MSG + " " + id);
+        }
+        return new ResponseEntity(endpoint, HttpStatus.ACCEPTED);
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('DELETE_ENDPOINT')")
+    public ResponseEntity<Void> deleteEndpoint(String id) {
+        try {
+            endpointService.deleteEndpoint(id);
+        } catch (EndpointService.EndpointNotFoundException e) {
+            throw new ResourceNotFoundException(NOT_FOUND_ERROR_MSG + " " + id);
+        } catch (ClientNotFoundException e) {
+            throw new ConflictException("Client not found for the given endpoint with id: " + id);
+        }
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('EDIT_OPENAPI3_ENDPOINT')")
+    public ResponseEntity<Endpoint> updateEndpoint(String id, Endpoint endpoint) {
+        Endpoint ep;
+        try {
+            ep = endpointConverter.convert(endpointService.updateEndpoint(id, endpoint));
+        } catch (EndpointService.EndpointNotFoundException e) {
+            throw new ResourceNotFoundException(NOT_FOUND_ERROR_MSG + " " + id);
+        } catch (EndpointService.IllegalGeneratedEndpointUpdateException e) {
+            throw new ConflictException("Updating is not allowed for generated endpoint " + id);
```

Comment:
Should this be `BadRequestException` instead? I think that trying to update a generated endpoint is the user doing something wrong rather than the system being in a conflicted state (such as having duplicate value or missing the parent object).